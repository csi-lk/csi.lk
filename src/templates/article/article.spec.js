import React from 'react'

import ArticleTemplate from './article'

describe('Template: Article', () => {
  let props
  beforeEach(() => {
    props = {
      data: {
        markdownRemark: {
          html: '<p>Test</p>',
          frontmatter: {
            title: 'Test',
            tags: 't1, t2',
            description: 'Test desc',
          },
        },
      },
      pathContext: {
        path: '/test',
      },
    }
  })
  it('renders', () => {
    const wrapper = shallow(<ArticleTemplate {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders preview', () => {
    const wrapper = shallow(<ArticleTemplate {...props} preview />)
    expect(wrapper).toMatchSnapshot()
  })
})
