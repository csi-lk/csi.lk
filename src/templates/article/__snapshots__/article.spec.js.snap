// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template: Article renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleTemplate
    data={
      Object {
        "markdownRemark": Object {
          "frontmatter": Object {
            "title": "Test",
          },
          "html": "<p>Test</p>",
        },
      }
    }
    pathContext={
      Object {
        "path": "/test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <meta
            charSet="utf-8"
          />
          <title>
            Test
          </title>
          <meta
            content={undefined}
            name="description"
          />
          <meta
            content="Test"
            name="og:title"
          />
          <meta
            content="https://www.csi.lk/test"
            name="og:url"
          />
          <meta
            content="http://www.csi.lk/into-code.png"
            name="og:image"
          />
          <meta
            content={undefined}
            name="og:description"
          />
        </HelmetWrapper>,
        <a
          href="/"
        >
          back home
        </a>,
        <article
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Test</p>",
            }
          }
        />,
      ],
      "className": "article",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <meta
              charSet="utf-8"
            />,
            <title>
              Test
            </title>,
            <meta
              content={undefined}
              name="description"
            />,
            <meta
              content="Test"
              name="og:title"
            />,
            <meta
              content="https://www.csi.lk/test"
              name="og:url"
            />,
            <meta
              content="http://www.csi.lk/into-code.png"
              name="og:image"
            />,
            <meta
              content={undefined}
              name="og:description"
            />,
          ],
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "charSet": "utf-8",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
            },
            "ref": null,
            "rendered": "Test",
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": undefined,
              "name": "description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "Test",
              "name": "og:title",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "https://www.csi.lk/test",
              "name": "og:url",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "http://www.csi.lk/into-code.png",
              "name": "og:image",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": undefined,
              "name": "og:description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "back home",
          "href": "/",
        },
        "ref": null,
        "rendered": "back home",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<p>Test</p>",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <meta
              charSet="utf-8"
            />
            <title>
              Test
            </title>
            <meta
              content={undefined}
              name="description"
            />
            <meta
              content="Test"
              name="og:title"
            />
            <meta
              content="https://www.csi.lk/test"
              name="og:url"
            />
            <meta
              content="http://www.csi.lk/into-code.png"
              name="og:image"
            />
            <meta
              content={undefined}
              name="og:description"
            />
          </HelmetWrapper>,
          <a
            href="/"
          >
            back home
          </a>,
          <article
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>Test</p>",
              }
            }
          />,
        ],
        "className": "article",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <meta
                charSet="utf-8"
              />,
              <title>
                Test
              </title>,
              <meta
                content={undefined}
                name="description"
              />,
              <meta
                content="Test"
                name="og:title"
              />,
              <meta
                content="https://www.csi.lk/test"
                name="og:url"
              />,
              <meta
                content="http://www.csi.lk/into-code.png"
                name="og:image"
              />,
              <meta
                content={undefined}
                name="og:description"
              />,
            ],
            "defer": true,
            "encodeSpecialCharacters": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "charSet": "utf-8",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
              },
              "ref": null,
              "rendered": "Test",
              "type": "title",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": undefined,
                "name": "description",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "Test",
                "name": "og:title",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "https://www.csi.lk/test",
                "name": "og:url",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "http://www.csi.lk/into-code.png",
                "name": "og:image",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": undefined,
                "name": "og:description",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "back home",
            "href": "/",
          },
          "ref": null,
          "rendered": "back home",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<p>Test</p>",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
